tree/ntuple/v7/src/RField.cxx

```diff
@@ -417,6 +421,26 @@ void ROOT::Experimental::Detail::RFieldBase::RemoveReadCallback(size_t idx)
    fIsSimple = (fTraits & kTraitMappable) && fReadCallbacks.empty();
 }
 
+void ROOT::Experimental::Detail::RFieldBase::AddReadCallbacksFromIORules(
+   const std::span<const ROOT::TSchemaRule *> rules, TClass *classp)
+{
+   for (const auto rule : rules) {
+      if (rule->GetRuleType() != ROOT::TSchemaRule::kReadRule) {
+         R__LOG_WARNING(NTupleLog()) << "ignoring I/O customization rule with unsupported type";
+         continue;
+      }
+      if (auto func = rule->GetReadFunctionPointer()) {
```

Comment:
What about the else branch? Should we warn when this happens? Or assert?

Reply:
Probably `R__ASSERT`, yes.  @pcanal may correct me, but if the rule type is `kReadRule`, there will always be an associated function.